### ISSUE TITLE
[HOLD for payment 2025-01-07] Infinite loading on downgrade button if a secondary admin tries to downgrade plan

### ISSUE THREAD

**Issue by mitarachim** [Expensify/App#54493]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.78-1
**Reproducible in staging?:** Yes
**Reproducible in production?:** No
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** N/A
**If this was caught during regression testing, add the test name, ID and link from TestRail:** Exp
**Email or phone of affected tester (no customers):** nathanmulugetatesting+2549@gmail.com
**Issue reported by:** Applause Internal Team
**Device used:** Windows 11 pro/chrome
**App Component:** Workspace Settings

## Action Performed:
1. Navigate to staging.new.expensify.com
2. Create a workspace
3. Invite a member and make the member an admin
4. As the admin go to the profile of the workspace
5. Upgrade the plan to a control workspace
6. Downgrade the plan to a collect workspace

## Expected Result:
Secondary admin is able to downgrade

## Actual Result:
Infinite loading on the downgrade button

## Workaround:
Unknown

## Platforms:
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/cb45ec16-fb5f-45c3-bf93-41fea9b9a037

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Issue Owner</summary>Current Issue Owner: @laurenreidexpensify</details>

**Comment by techievivek** [comment]
Seems like a backend bug to me, just finishing up my PR for Hubspot integration then I can take a look into it.

**Comment by DylanDylann** [comment]
Agree that it is a BE problem. But we also need to adjust a minor thing in the FE, I have created a PR to do that

**Comment by techievivek** [comment]
Ok, just tested this flow and this is the backend response I am seeing.

<img width="1680" alt="Screenshot 2024-12-24 at 5 57 14 PM" src="https://github.com/user-attachments/assets/1e948a16-3030-4935-a39a-aa8b046620b6" />

**Comment by techievivek** [comment]
@carlosmiceli Are you working on the design doc for this feature? I think we should probably introduce a new command since we need to use Onyx::handleException here https://github.com/Expensify/Web-Expensify/blob/6b67a5831630e274a3161633a7113c4517fa0de6/lib/PolicyAPI.php#L5653, since we need to populate the error data in onyx.

**Comment by techievivek** [comment]
I'm not sure if there has been any discussion about introducing a new command for the newDot use case. I'll go ahead and assign this to you, @carlosmiceli. Let me know if you need any help with this.

**Comment by carlosmiceli** [comment]
Yes, you're right about everything you mention here @techievivek. We decided to adjust the current command, but it was only when we hooked up the API that we noticed the multiple Auth calls as well. It's next on my list 🙏

**Comment by carlosmiceli** [comment]
@techievivek I'm realizing that it may take a while before I can build the new command, so I wonder if you agree that we could implement some temporary FE fix to deal with the error response?

**Comment by laurenreidexpensify** [comment]
@carlosmiceli @techievivek it looks like @jayeshmangwani and @DylanDylann did work on this PR - can you confirm if any payment is due here? Thanks

**Comment by jayeshmangwani** [comment]
@laurenreidexpensify There’s no payment due for us at the moment, as this is part of the workspace downgrade project and payment will be handled collectively at the end.

**Comment by laurenreidexpensify** [comment]
Okay great - am closing this issue out then. Thanks !
